internal/clients/http/ec2/ec2_client.go

```diff
@@ -464,17 +554,63 @@ func (c *ec2Client) listAttachedRolePolicies(ctx context.Context, auth *clients.
 	return result, nil
 }
 
+func (c *ec2Client) listPoliciesFromAttached(ctx context.Context, policies []*iamTypes.AttachedPolicy) ([]*iamTypes.Policy, error) {
+	logger := logger(ctx)
+	logger.Info().Msgf("Fetching policies for attached policies.")
+	result := make([]*iamTypes.Policy, len(policies))
+	for i := range policies {
+		input := &iam.GetPolicyInput{
+			PolicyArn: policies[i].PolicyArn,
+		}
+		output, err := c.iam.GetPolicy(ctx, input)
+		if err != nil {
+			return nil, fmt.Errorf("cannot get policy for policy arn %s: %w", *input.PolicyArn, err)
+		}
+		result[i] = output.Policy
+
+	}
+	return result, nil
+}
+
+func (c *ec2Client) listPolicyVersions(ctx context.Context, policies []*iamTypes.Policy) ([]*iamTypes.PolicyVersion, error) {
+	result := make([]*iamTypes.PolicyVersion, len(policies))
+	for i := range policies {
+		input := &iam.GetPolicyVersionInput{
+			PolicyArn: policies[i].Arn,
+			VersionId: policies[i].DefaultVersionId,
+		}
+		output, err := c.iam.GetPolicyVersion(ctx, input)
+		if err != nil {
+			return nil, fmt.Errorf("cannot get policy version for policy arn %s, version %s: %w", *input.PolicyArn, *input.VersionId, err)
+		}
+		result[i] = output.PolicyVersion
+
+	}
+	return result, nil
+}
+
 func (c *ec2Client) CheckPermission(ctx context.Context, auth *clients.Authentication) ([]string, error) {
 	logger := logger(ctx)
 	logger.Info().Msgf("Listing policies attached to the role.")
-	_, err := c.listAttachedRolePolicies(ctx, auth)
+
+	attachedRolePolicies, err := c.listAttachedRolePolicies(ctx, auth)
 	if err != nil {
 		return nil, err
 	}
+	policies, err := c.listPoliciesFromAttached(ctx, attachedRolePolicies)
+	if err != nil {
+		return nil, fmt.Errorf("Could not get policies: %w", err)
+	}
 
-	// TODO: GetPolicies for listed policies.
+	versions, err := c.listPolicyVersions(ctx, policies)
+	if err != nil {
+		return nil, fmt.Errorf("Could not get policy versions: %w", err)
+	}
 
-	// TODO: GetPolicyVersion for policies.
+	statements, err := listStatements(versions)
+	if err != nil {
+		return nil, fmt.Errorf("Could not list statements: %w", err)
+	}
 
-	return nil, nil
+	return listMissingPermissions(statements, expectedStatement), nil
 }
```

Comment:
It looks great, I would maybe consider moving this from already crowded `ec2_client.go` into a separate file.

Reply:
if we are moving this from the ec2_client.go file I would expect to have an EC2 folder with all the relevant related functions. And the functions with shared logic are stored in the same file.